src/main/resources/com/google/api/codegen/java/test.snip

```diff
@@ -42,15 +43,97 @@
       api.close();
     }
 
-    @join case : xapiTest.testCases
-      {@testCase(case)}
+    @join test : xapiTest.testCases
+      {@testCase(test)}
+
     @end
   }
 @end
 
-@private testCase(case)
+@private testCase(test)
   @@Test
   @@SuppressWarnings("all")
-  public void {@case.name}() {
+  public void {@test.name}() {
+    {@initCode(test.initCode)}
+    {@methodCall(test)}
+
+    List<GeneratedMessage> requests = serviceHelper.getService().getRequests();
+    Assert.assertEquals(1, requests.size());
+    {@test.requestType} request = ({@test.requestType})requests[0];
+
+    @join assert : test.asserts
+      Assert.assertEquals(request.{@assert.actualValueIdentifier}(), {@assert.expectedValueIdentifier});
+    @end
   }
 @end
+
+@private methodCall(test)
+  @if {@test.isPageStreaming}
+    PageAccessor<{@test.resourceType}> response = api.{@test.methodName}(\
+      {@sampleMethodCallArgList(test.initCode.fieldSettings)});
+    for ({@test.resourceType} element : response) {
+      // DO NOTHING
+    }
+  @else
+    api.{@test.methodName}(\
+      {@sampleMethodCallArgList(test.initCode.fieldSettings)});
+  @end
+@end
+
+@private sampleMethodCallArgList(fieldSettings)
+  @join fieldSetting : fieldSettings on ", "
+    {@fieldSetting.identifier}
+  @end
+@end
+
+@private initCode(initCodeSpec)
```

Comment:
sounds good. or you don't have to worry about this right now, I can figure out how to share and reuse after your first MVVM submission is done.